cli/command/stack/kubernetes/list.go

```diff
@@ -44,6 +54,56 @@ func getStacks(kubeCli *KubeCli, opts options.List) ([]*formatter.Stack, error)
 	return formattedStacks, nil
 }
 
+func getStacksWithAllNamespaces(dockerCli command.Cli, kubeCli *KubeCli, opts options.List) ([]*formatter.Stack, error) {
+	stacks, err := getStacks(kubeCli, opts)
+	if err == nil || !apierrs.IsForbidden(err) {
+		return stacks, err
+	}
+	nms, err2 := getUserVisibleNamespaces(dockerCli)
+	if err2 != nil {
+		logrus.Debugf("Failed to query user visible namespaces: %s", err2)
+		// ignore this last error and return the previous one instead
+		return nil, err
+	}
+	opts.AllNamespaces = false
+	for _, namespace := range nms.Items {
+		kubeCli.kubeNamespace = namespace.Name
+		ss, err := getStacks(kubeCli, opts)
+		if err != nil {
+			return nil, err
+		}
+		stacks = append(stacks, ss...)
+	}
+	return stacks, nil
+}
+
+func getUserVisibleNamespaces(dockerCli command.Cli) (*core_v1.NamespaceList, error) {
+	host := dockerCli.Client().DaemonHost()
+	endpoint, err := url.Parse(host)
+	if err != nil {
+		return nil, err
+	}
+	endpoint.Scheme = "https"
```

Comment:
This part is a bit sad as this is something we already do in the `moby/moby` `client` package (setting the scheme to `https` when tls is configured)..